### ISSUE TITLE
Question about the proposal_boxes in the code

### ISSUE THREAD

**Issue by suyeH** [PeizeSun/SparseR-CNN#92]
Hi, thanks for your excellent work. After reading the code, I have some questions about the updating of proposal_boxes.

1.  clone() is used here, so why will the weight of `self.init_proposal_boxes` be updated through back propagation? 
 [`proposal_boxes = self.init_proposal_boxes.weight.clone()`](https://github.com/PeizeSun/SparseR-CNN/blob/29cf98224edab842086ff19f97a9000029af8fe7/projects/SparseRCNN/sparsercnn/detector.py#L128)

2. Why detach() is necessary here?
[bboxes = pred_bboxes.detach()](https://github.com/PeizeSun/SparseR-CNN/blob/29cf98224edab842086ff19f97a9000029af8fe7/projects/SparseRCNN/sparsercnn/head.py#L109)

**Comment by PeizeSun** [comment]
Hi~
1. clone won't block the gradient. (In fact, clone is an unnecessary op here, it can be removed.)
2. we find detach the gradient here benefits to final performance.

**Comment by suyeH** [comment]
> Hi~
> 
> 1. clone won't block the gradient. (In fact, clone is an unnecessary op here, it can be removed.)
> 2. we find detach the gradient here benefits to final performance.

I got it. Thanks for your reply.
